src/mapping/query/functions.rs

```diff
@@ -0,0 +1,444 @@
+        Self { query, default }
+    }
+}
+
+impl Function for ToIntegerFn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let result = match self.query.execute(ctx) {
+            Ok(v) => match v {
+                Value::Integer(_) => Ok(v),
+                Value::Float(f) => Ok(Value::Integer(f as i64)),
+                Value::Bytes(_) => Conversion::Integer.convert(v).map_err(|e| e.to_string()),
+                Value::Boolean(b) => Ok(Value::Integer(if b { 1 } else { 0 })),
+                Value::Timestamp(t) => Ok(Value::Integer(t.timestamp())),
+                Value::Null => Err("value is null".to_string()),
+                _ => Err("unable to convert array or object into int".to_string()),
+            },
+            Err(err) => Err(err),
+        };
+        if result.is_err() {
+            if let Some(v) = &self.default {
+                return Ok(v.clone());
+            }
+        }
+        result
+    }
+}
+
+//------------------------------------------------------------------------------
+
+#[derive(Debug)]
+pub(in crate::mapping) struct ToFloatFn {
+    query: Box<dyn Function>,
+    default: Option<Value>,
+}
+
+impl ToFloatFn {
+    pub(in crate::mapping) fn new(query: Box<dyn Function>, default: Option<Value>) -> Self {
+        Self { query, default }
+    }
+}
+
+impl Function for ToFloatFn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let result = match self.query.execute(ctx) {
+            Ok(v) => match v {
+                Value::Float(_) => Ok(v),
+                Value::Integer(i) => Ok(Value::Float(i as f64)),
+                Value::Bytes(_) => Conversion::Float.convert(v).map_err(|e| e.to_string()),
+                Value::Boolean(b) => Ok(Value::Float(if b { 1.0 } else { 0.0 })),
+                Value::Timestamp(t) => Ok(Value::Float(t.timestamp() as f64)),
+                Value::Null => Err("value is null".to_string()),
+                _ => Err("unable to convert array or object into float".to_string()),
+            },
+            Err(err) => Err(err),
+        };
+        if result.is_err() {
+            if let Some(v) = &self.default {
+                return Ok(v.clone());
+            }
+        }
+        result
+    }
+}
+
+//------------------------------------------------------------------------------
+
+#[derive(Debug)]
+pub(in crate::mapping) struct ToBooleanFn {
+    query: Box<dyn Function>,
+    default: Option<Value>,
+}
+
+impl ToBooleanFn {
+    pub(in crate::mapping) fn new(query: Box<dyn Function>, default: Option<Value>) -> Self {
+        Self { query, default }
+    }
+}
+
+impl Function for ToBooleanFn {
+    fn execute(&self, ctx: &Event) -> Result<Value> {
+        let result = match self.query.execute(ctx) {
+            Ok(v) => match v {
+                Value::Boolean(_) => Ok(v),
+                Value::Float(f) => Ok(Value::Boolean(f > 0.0)),
+                Value::Integer(i) => Ok(Value::Boolean(i > 0)),
```

Comment:
The more expected convention from other languages would be `!= 0` instead of `> 0`.